### PR TITLE
Replaced BookRepository with JpaRepository, added Liquibase, implemented soft delete, and update method.

### PR DESCRIPTION
**Pull request by noiseintheattic**
Replaced BookRepository with JpaRepository, 
added Liquibase, 
implemented soft delete, and update method.

### CODE REVIEW

File: src/main/java/mate/academy/controller/BookController.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(value = "/api/books")
+public class BookController {
+
```
Comment (George-Khomiak) [inline src/main/java/mate/academy/controller/BookController.java]: Don't start a class implementation with an empty line

File: src/main/java/mate/academy/dto/BookDto.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package mate.academy.dto;
+
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Data
+public class BookDto {
+
```
Comment (George-Khomiak) [inline src/main/java/mate/academy/dto/BookDto.java]: empty line

File: src/main/java/mate/academy/dto/CreateBookRequestDto.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package mate.academy.dto;
+
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Data
+public class CreateBookRequestDto {
+
```
Comment (George-Khomiak) [inline src/main/java/mate/academy/dto/CreateBookRequestDto.java]: two empty lines here

File: src/main/java/mate/academy/mapper/BookMapper.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+import mate.academy.config.MapperConfig;
+import mate.academy.dto.BookDto;
+import mate.academy.dto.CreateBookRequestDto;
+import mate.academy.model.Book;
+import org.mapstruct.Mapper;
+import org.mapstruct.MappingTarget;
+
+@Mapper(config = MapperConfig.class)
+public interface BookMapper {
+
```
Comment (George-Khomiak) [inline src/main/java/mate/academy/mapper/BookMapper.java]: again

File: src/main/java/mate/academy/service/BookServiceImpl.java [hunk 5]
```diff
@@ -1,33 +1,63 @@
...
+    public BookDto update(Long id, CreateBookRequestDto createBookRequestDto) {
+        Book book = bookRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException("Can't find book with id: " + id));
+
+        bookMapper.updateFromDto(createBookRequestDto, book);
+        Book updatedBook = bookRepository.save(book);
+
+        return bookMapper.toDto(updatedBook);
     }
+
```
Comment (George-Khomiak) [inline src/main/java/mate/academy/service/BookServiceImpl.java]: --

**Review by George-Khomiak — CHANGES_REQUESTED** [review]
Very good work, logic is in place, just remove all the empty lines in places where they don't improve readability

File: src/main/java/mate/academy/controller/BookController.java (hunk 1 above)
Comment (noiseintheattic) [reply src/main/java/mate/academy/controller/BookController.java]: ok, done

File: src/main/java/mate/academy/dto/BookDto.java (hunk 2 above)
Comment (noiseintheattic) [reply src/main/java/mate/academy/dto/BookDto.java]: ok, done

File: src/main/java/mate/academy/mapper/BookMapper.java (hunk 4 above)
Comment (noiseintheattic) [reply src/main/java/mate/academy/mapper/BookMapper.java]: ok, done

File: src/main/java/mate/academy/service/BookServiceImpl.java (hunk 5 above)
Comment (noiseintheattic) [reply src/main/java/mate/academy/service/BookServiceImpl.java]: ok, done

File: src/main/java/mate/academy/dto/CreateBookRequestDto.java (hunk 3 above)
Comment (noiseintheattic) [reply src/main/java/mate/academy/dto/CreateBookRequestDto.java]: ok, done

**Review by George-Khomiak — APPROVED** [review]

**Pull request closed without merging**